### PR TITLE
update mailmap and add CI check

### PR DESCRIPTION
**Pull request by renefritze**
Compared to #1491 this drops adding mailmap updates from release checklist,
and adds a gitlab CI check instead.

### CODE REVIEW

**Comment by pmli** [comment]
Can this detect when both the name and email change (as with Josefine)?

**Comment by renefritze** [comment]
> Can this detect when both the name and email change (as with Josefine)?

Hmm, no. It also doesn't catch "same email - different names". Which is easy enough to add. But for Josefine's case it's harder to detect. Would need some fuzzier matching of the name I guess, which I have very little experience with.

**Comment by renefritze** [comment]
I think this can now be merged instead of #1491, @pmli

File: .mailmap [hunk 1]
```diff
@@ -12,6 +20,9 @@ Felix Schindler <felix.schindler@wwu.de> <mail@felixalbrecht.de>
 Geordie McBain <gdmcbain@protonmail.com>
 Hendrik Kleikamp <hendrik.kleikamp@uni-muenster.de>
 Hendrik Kleikamp <hendrik.kleikamp@uni-muenster.de> <h_klei18@uni-muenster.de>
+Josefine Zeller <josefine_zeller@web.de> josefinez <josefine.zeller@web.de>
+Jonas Nicodemus <jonas.nicodemus@icloud.com>
+Jonas Nicodemus <jonas.nicodemus@icloud.com> <44996273+Jonas-Nicodemus@users.noreply.github.com>
```
Comment (pmli) [inline .mailmap:25]: It would be good to sort this alphabetically.

File: docs/RELEASE_CHECKLIST.md [hunk 2]
```diff
@@ -27,5 +27,6 @@
  1. [ ] Add a new section in
         [`docker/docs/releases/Dockerfile`](https://github.com/pymor/docker/blob/main/docs/releases/Dockerfile).
  1. [ ] All developers check if (stale) branches can be pruned.
+ 1. [ ] All developers check if for mailmap correctness.
```
Comment (pmli) [inline docs/RELEASE_CHECKLIST.md:30]:
```suggestion
 1. [ ] All developers check for mailmap correctness.
```

**Review by pmli — CHANGES_REQUESTED** [review]

**Comment by renefritze** [comment]
Thanks @pmli! All fixed now.

**Review by pmli — APPROVED** [review]

**Pull request merged**